### ISSUE TITLE
Petname generation is not "random" unless rand.Seed is set

### ISSUE THREAD

**Issue by jdhayford** [dustinkirkland/golang-petname#10]
This is fairly obvious after taking a peek at the source and seeing the use of `rand.Intn`. That said I think it would be a good idea to add a note in the Readme about this and recommend setting the seed before generating any petnames using the library. If this is the case and I'm not missing something I'd be happy to spin up a PR.

Thanks for the library!

**Comment by dustinkirkland** [comment]
Good call!  I've added it to the README.md sample golang code.  If you have
other ideas, sure, go ahead and send a PR.

Cheers,
@DustinKirkland

**Comment by romainverduci** [comment]
Hi !
I'm using the lib for test automation and I'd like to have a "non deterministic" mode so when the tests run multiple times they don't produce the same test data.
@dustinkirkland I'm happy to submit a PR for this mode but is that something you would like the lib to support ?

Cheers :)

**Comment by dustinkirkland** [comment]
Sure, I'd be happy to take a look!  Thanks!

@DustinKirkland

**Comment by dustinkirkland** [comment]
This issue has been resolved by Go's evolution! ✓

**What changed:**
Since Go 1.20 (released Feb 2023), the `rand` package is automatically seeded at program startup. This means petname generation is now random by default without any manual seeding required.

From Go 1.20 release notes:
> The top-level functions now use a global random generator that is automatically seeded

**Current state:**
- This package requires Go 1.20+ (see go.mod)
- Random generation works out of the box
- No need to call `rand.Seed()` or `petname.NonDeterministicMode()`

**Note:** The README examples still show `rand.Seed()` calls, but these are now unnecessary in Go 1.20+. This is tracked in issue #15.

Thanks for reporting this back in 2019 - it helped improve the library, and Go's improvements have now made it even better!
